v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/cache.ts

```diff
@@ -0,0 +1,74 @@
+import path from "node:path";
+
+import {
+  exists,
+  getAccessTime,
+  getAllFilesMatching,
+  getFileSize,
+  readJsonFile,
+  remove,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+
+export class ObjectCache<T> {
+  readonly #path: string;
+  readonly #defaultMaxAgeMs: number;
+  readonly #defaultMaxSize: number;
+
+  constructor(
+    basePath: string,
+    namespace: string,
+    version: string,
+    defaultMaxAgeMs: number = 7 * 24 * 60 * 60 * 1000, // 1 week
+    defaultMaxSize: number = 2 * 1024 * 1024 * 1024, // 2 GB
+  ) {
+    this.#path = path.join(basePath, namespace, version);
+    this.#defaultMaxAgeMs = defaultMaxAgeMs;
+    this.#defaultMaxSize = defaultMaxSize;
+  }
+
+  public async set(key: string, value: T): Promise<void> {
+    const filePath = path.join(this.#path, key);
```

Comment:
It's unclear to me if these files have any extension. I think I would add `.json` to play safe.